code/modules/clothing/masks/gas_filter.dm

```diff
@@ -0,0 +1,113 @@
+///List of gases with high filter priority
+GLOBAL_LIST_INIT(high_filtering_gases, list(
+	/datum/gas/plasma,
+	/datum/gas/carbon_dioxide,
+	/datum/gas/nitrous_oxide
+))
+///List of gases with medium filter priority
+GLOBAL_LIST_INIT(mid_filtering_gases, list(
+	/datum/gas/nitryl,
+	/datum/gas/stimulum,
+	/datum/gas/freon,
+	/datum/gas/hypernoblium,
+	/datum/gas/bz
+))
+///List of gases with low filter priority
+GLOBAL_LIST_INIT(low_filtering_gases, list(
+	/datum/gas/healium,
+	/datum/gas/proto_nitrate,
+	/datum/gas/halon,
+	/datum/gas/tritium,
+	/datum/gas/zauker
+))
+
+///Filtering ratio for high amounts of gas
+#define HIGH_FILTERING_RATIO 0.001
+///Filtering ratio for min amount of gas
+#define LOW_FILTERING_RATIO 0.0005
+///Min amount of high filtering gases for high filtering ratio
+#define HIGH_FILTERING_MOLES 0.001
+///Min amount of mid filtering gases for high filtering ratio
+#define MID_FILTERING_MOLES 0.0025
+///Min amount of low filtering gases for high filtering ratio
+#define LOW_FILTERING_MOLES 0.0005
+///Min amount of wear that the filter gets when used
+#define FILTERS_CONSTANT_WEAR 0.05
```

Comment:
How much do these offset things like taking damage from plasma/sleeping due to n2o?

Are plasmamen fucked?

Reply:
i'm going to add a plasmaman specific filter that removes O2 instead of plasma. These are done so that the gas get removed from the breathed gas, if the percentage is high in the breath you'll still get the effects since a bit of gas pass.
For example: a room has 100 moles total, out of these 70 are N2, 29 are O2 and 1 is plasma. 
Now if we breathe in that air, we'll get: 
breath% = breath_volume / cell_volume = 1.99 / 2500 = 0.000796 %
breath_moles = cell_moles * breath% = 100 * 0.000796 = 0.0769 moles
so out of these moles we'll have 
70% N2 = 0.05572 moles of N2
29% O2 = 0.023084 moles of O2
1% Plasma = 0.000796 moles of plasma
the gases pass through the filter, but since the plasma moles (that has a high filtering efficience) are lower than HIGH_FILTERING_MOLES, all of them will be removed from the breath gases and no effect will be had on the mob.
BUT if the plasma is higher, something like 6%, the moles will be 0.004776 moles, this number is higher than the defined value. In this case we'll still remove some moles, but the remaining ones will be breathed in and cause the effects on the mob, lessened by the removal of a portion of that gas (for example n2o could be filtered and the breathed gas will not reach the sleep threshold but just the giggle one)

Reply:
Sounds good